commerce/domain/shipping/splitshipment/SplitShipmentSample.cls

```diff
@@ -0,0 +1,198 @@
+/**
+ * @description Custom Split Shipment Service sample.
+ */
+global class SplitShipmentSample extends CartExtension.SplitShipmentService {
+
+    /**
+     * @description All classes extending CartExtension.SplitShipmentService must have a default constructor defined.
+     */
+    global SplitShipmentSample() {}
+
+    /**
+     * @description Constructor used by unit tests only. See <<SplitShipmentUnitTest>>.
+     * @param mockService Mocked service which can be used to stub the default Split Shipment logic.
+     */
+    global SplitShipmentSample(final CartExtension.SplitShipmentServiceMock mockService) {
+        super(mockService);
+    }
+
+    public virtual override void arrangeItems(CartExtension.ItemArrangementRequest request) {
+        CartExtension.Cart cart = request.getCart();
+        List<CartExtension.ItemArrange> itemArrangeList = request.getItemArrangeList();
+
+        for (CartExtension.ItemArrange itemArrange : itemArrangeList) {
+            CartExtension.CartDeliveryGroup cartDeliveryGroup = findOrCreateDeliveryGroup(cart, itemArrange);
+            updateOrCreateCartItemToCartDeliveryGroup(cart, cartDeliveryGroup, itemArrange);
+        }
+    }
+
+    /**
+     * @description Either finds a Cart Delivery Group based on the given inputs or creates a new one.
+     * @param cart Representation of a `WebCart` record
+     * @param itemArrange Holds details of how to arrange an item between delivery groups
+     */
+    private CartExtension.CartDeliveryGroup findOrCreateDeliveryGroup(CartExtension.Cart cart, CartExtension.ItemArrange itemArrange) {
+        Address deliveryAddress = itemArrange.getDeliveryAddress();
+
+        if (itemArrange.getDeliveryGroupId() != null) {
+            Iterator<CartExtension.CartDeliveryGroup> cdgIterator = cart.getCartDeliveryGroups().iterator();
+            while (cdgIterator.hasNext()) {
```

Comment:
so, we wanted to lookup by Delivery Group by id, but there is no such Delivery Group found. Should we terminate with an error in that case? Why are we proceeding?